lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,393 @@
+		}
+
+		$schema       = $this->get_item_schema();
+		$extra_fields = array(
+			'name',
+			'category',
+			'editor_script',
+			'script',
+			'editor_style',
+			'style',
+			'supports',
+		);
+		foreach ( $extra_fields as $extra_field ) {
+			if ( rest_is_field_included( $extra_field, $fields ) ) {
+				$field                = isset( $block_type->$extra_field ) ? $block_type->$extra_field : '';
+				$data[ $extra_field ] = rest_sanitize_value_from_schema( $field, $schema['properties'][ $extra_field ] );
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $block_type ) );
+
+		/**
+		 * Filters a block type returned from the REST API.
+		 *
+		 * Allows modification of the block type data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response   The response object.
+		 * @param object           $block_type The original block type object.
+		 * @param WP_REST_Request  $request    Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_block_type', $response, $block_type, $request );
+	}
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @param WP_Block_Type $block_type block type data.
+	 * @return array Links for the given block type.
+	 */
+	protected function prepare_links( $block_type ) {
+		$pieces    = explode( '/', $block_type->name );
+		$namespace = $pieces[0];
+		$links     = array(
+			'collection' => array(
+				'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+			),
+			'self'       => array(
+				'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $block_type->name ) ),
+			),
+			'up'         => array(
+				'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $namespace ) ),
+			),
+		);
+
+		if ( $block_type->is_dynamic() ) {
+			$links['https://api.w.org/render-block']['href'] = add_query_arg( 'context', 'edit', rest_url( sprintf( '%s/%s/%s', 'wp/v2', 'block-renderer', $block_type->name ) ) );
+		}
+
+		return $links;
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block-type',
+			'type'       => 'object',
+			'properties' => array(
+				'name'          => array(
+					'description' => __( 'Unique name identifying the block type.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'attributes'    => array(
```

Comment:
@gziolo should we add parent here. And it is okay to default to an empty array.

Reply:
Let's include all fields as defined in RFC:
https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#block-api

`parent` is optional, so it shouldn't default to an empty array.

There are also:
- `title`
- `icon`
- `description`
- `keywords`
- `styles` – `styleVariations` is an alias used in `block.json`

Reply:
@gziolo What about context?

Reply:
```json
{"attributes":{"align":{"type":"string","enum":["left","center","right","wide","full"]},"className":{"type":"string"},"displayAsDropdown":{"type":"boolean","default":false},"showPostCounts":{"type":"boolean","default":false},"layout":{"type":"string"}},"is_dynamic":true,"name":"core\/archives","category":"widgets","editor_script":"","script":"","editor_style":"","style":"","supports":{"align":true,"html":false},"title":"","icon":"","description":"","keywords":[],"parent":[],"styles":[],"text_domain":"","_links":{"collection":[{"href":"http:\/\/localhost:8889\/index.php?rest_route=\/__experimental\/block-types"}],"self":[{"href":"http:\/\/localhost:8889\/index.php?rest_route=\/__experimental\/block-types\/core\/archives"}],"up":[{"href":"http:\/\/localhost:8889\/index.php?rest_route=\/__experimental\/block-types\/core"}],"wp:render-block":[{"href":"http:\/\/localhost:8889\/index.php?rest_route=%2Fwp%2Fv2%2Fblock-renderer%2Fcore%2Farchives&context=edit"}],"curies":[{"name":"wp","href":"https:\/\/api.w.org\/{rel}","templated":true}]}}
```

Updated the fields. Here is an example output from the api. @TimothyBJacobs

Reply:
I think both `context` and `providesContext`. I’m on mobile, @aduth can you help. It isn’t included in RFC for block registration.

Reply:
👍 thanks @spacedmonkey! Do we have output when a plugin has those additional fields in use?

Reply:
Not yet as it up to @gziolo and the rest of the gutenberg team to get those fields registered in the block.json files.

Reply:
I have added 

- parent
- title
- icon
- description
- keywords
- styles

and added tests for them.

I think for now, we should adding `context` and `providesContext`. It is super easy to add these later after they have been added to the RFC and are more widely used. I think we may want be able to filter by context, so renaming this would useful as context as a param is always in use.

Reply:
> I think for now, we should adding context and providesContext. It is super easy to add these later

Sorry, to clarify, you're saying we should _skip_ adding `context` and `providesContext` for now?

Reply:
> Sorry, to clarify, you're saying we should skip adding context and providesContext for now?

Yes, skip / not include them as they are not cofirmed / part of the RFC.

Reply:
:+1: works for me.